### PR TITLE
Unit test for accountsNewRound creatables parameter

### PR DESCRIPTION
**Pull request by algonautshant**
Added random sampling for creatables, and compared them to the database image in every iteration.

### CODE REVIEW

**Comment by algonautshant** [comment]
Looks like this overlaps with https://github.com/algorand/go-algorand/pull/1349

File: ledger/accountdb_test.go [hunk 1]
```diff
@@ -356,19 +356,134 @@ func TestAccountDBRound(t *testing.T) {
 	require.NoError(t, err)
 	checkAccounts(t, tx, 0, accts)
 
+	ctbsList, randomCtbs := randomCreatables()
+	expectedDbImage := make(map[basics.CreatableIndex]modifiedCreatable)
+	
 	for i := 1; i < 10; i++ {
+		ctbsWithDeletes := randomCreatableSampling(i, ctbsList, randomCtbs, expectedDbImage)
```
Comment (algorandskiy) [inline ledger/accountdb_test.go:363]: looks like it conflicts with my PR #1349 - please rebase

File: ledger/accountdb_test.go [hunk 2]
```diff
@@ -356,19 +356,134 @@ func TestAccountDBRound(t *testing.T) {
...
+// loop 1: returns: * the second 10 elements
+//                  * random sample of elements from the first 10: created changed from true -> false
+// loop 2: returns: * the elements 20->30
+//                  * random sample of elements from 10->20: created changed from true -> false
+func randomCreatableSampling(iteration int, crtbsList[]basics.CreatableIndex,
+	creatables map[basics.CreatableIndex]modifiedCreatable,
+	expectedDbImage map[basics.CreatableIndex]modifiedCreatable) (
+	map[basics.CreatableIndex]modifiedCreatable) {
+
+	numElementsPerSegement := 10
```
Comment (tsachiherman) [inline ledger/accountdb_test.go:421]: could you move the `numElementsPerSegement` to be an input parameter to the function ?

File: ledger/accountdb_test.go [hunk 3]
```diff
@@ -356,19 +356,134 @@ func TestAccountDBRound(t *testing.T) {
...
+
+	for i := delSegmentStart; i < delSegmentEnd; i++ {
+		ctb := creatables[crtbsList[i]]
+		if ctb.created == true {
+			if 1 == (crypto.RandUint64() % 2) {
+				ctb.created = false
+				newSample[crtbsList[i]] = ctb
+				delete(expectedDbImage, crtbsList[i])
+			}
+		}
```
Comment (tsachiherman) [inline ledger/accountdb_test.go:441]: could you merge the two if statements ? 
if created == true && 1 == (crypto..)

File: ledger/accountdb_test.go [hunk 4]
```diff
@@ -356,19 +356,134 @@ func TestAccountDBRound(t *testing.T) {
...
+		counter++
+		mc := modifiedCreatable{}
+		var buf[]byte
+		var asset basics.CreatableIndex
+		err := rows.Scan(&asset, &buf, &mc.ctype)
+		require.NoError(t, err)
+		copy(mc.creator[:], buf)
+
+		require.NotNil(t, expectedDbImage[asset])
+		require.Equal(t, mc.creator, expectedDbImage[asset].creator)
```
Comment (tsachiherman) [inline ledger/accountdb_test.go:402]: swap the order of the Equal parameters : the first one is supposed to be the "expected" and the latter is the actual value.

File: ledger/accountdb_test.go [hunk 5]
```diff
@@ -356,19 +356,134 @@ func TestAccountDBRound(t *testing.T) {
...
+		require.Equal(t, mc.ctype, expectedDbImage[asset].ctype)
+		require.Equal(t, expectedDbImage[asset].created, true)
```
Comment (tsachiherman) [inline ledger/accountdb_test.go:404]: When comparing to true, use `require.True(t, expectedDbImage[asset].created)`

File: ledger/accountdb_test.go [hunk 6]
```diff
@@ -356,19 +356,134 @@ func TestAccountDBRound(t *testing.T) {
...
+		ctype = basics.AppCreatable
+	}
+
+	creatable := modifiedCreatable{
+		ctype:   ctype,
+		created: (crypto.RandUint64() % 2) == 1,
+		creator: randomAddress(),
+		ndeltas: 1,
+	}
+	assetIdx := basics.CreatableIndex(crypto.RandUint64() % (uint64(2) << 50))
```
[the remark this answers is not in the document]
Comment (algonautshant) [reply ledger/accountdb_test.go:483]: Yes. I wanted to take my chances... fixed

File: ledger/accountdb_test.go (hunk 2 above)
Comment (algonautshant) [reply ledger/accountdb_test.go:421]: done.

File: ledger/accountdb_test.go (hunk 3 above)
Comment (algonautshant) [reply ledger/accountdb_test.go:441]: done

File: ledger/accountdb_test.go (hunk 4 above)
Comment (algonautshant) [reply ledger/accountdb_test.go:402]: fixed

File: ledger/accountdb_test.go (hunk 5 above)
Comment (algonautshant) [reply ledger/accountdb_test.go:404]: done

**Comment by tsachiherman** [comment]
currently failing on travis with :
```
--- FAIL: TestAccountDBRound (0.09s)
    accountdb_test.go:479: 
        	Error Trace:	accountdb_test.go:479
        	Error:      	Received unexpected error:
        	            	UNIQUE constraint failed: assetcreators.asset
        	Test:       	TestAccountDBRound
FAIL
```

**Comment by algorandskiy** [comment]
`knownCreatables` used to track ids from random deltas to get rid of duplicates in new random creatables:
```go
creatables := creatablesFromUpdates(updates, knownCreatables)
```

**Comment by algonautshant** [comment]
$ go test ./ledger/... -run TestAccountDBRound -cover -count=1
ok  	github.com/algorand/go-algorand/ledger	0.595s	coverage: 4.5% of statements

**Pull request merged**